linker/shared/src/main/scala/org/scalajs/linker/checker/ClassDefChecker.scala

```diff
@@ -704,10 +719,12 @@ private final class ClassDefChecker(classDef: ClassDef,
       case StoreModule() =>
         if (!classDef.kind.hasModuleAccessor)
           reportError(i"Illegal StoreModule inside class of kind ${classDef.kind}")
-        if (!env.inConstructor)
+        else if (!env.inConstructor)
           reportError(i"Illegal StoreModule outside of constructor")
-        if (env.thisType == NoType) // can happen before JSSuperConstructorCall in JSModuleClass
+        else if (env.thisType == NoType) // can happen before JSSuperConstructorCall in JSModuleClass
           reportError(i"Cannot find `this` in scope for StoreModule()")
+        else if (env.isThisRestricted)
```

Comment:
Why the change to else? Wouldn't we want to report errors exhaustively (if sensible)?

Reply:
I figured some of these errors might lead to other errors. But I reverted to not having `else`s. Same below.